From 7e55f5458c0e9fa7e51b36d97870e00e5fed8bb8 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Mon, 2 Jun 2014 17:08:20 -0400
Subject: [md] dm-cache: fix writethrough mode quiescing in cache_map

Message-id: <1401728901-921-3-git-send-email-snitzer@redhat.com>
Patchwork-id: 81835
O-Subject: [RHEL7.1 PATCH 2/3] dm cache: fix writethrough mode quiescing in cache_map
Bugzilla: 1103836
RH-Acked-by: Heinz Mauelshagen <heinzm@redhat.com>
RH-Acked-by: Joe Thornber <thornber@redhat.com>

BZ: 1103836

Upstream commit 131cd131a9ff63d4b84f3fe15073a2984ac30066
Author: Mike Snitzer <snitzer@redhat.com>
Date:   Thu May 1 16:14:24 2014 -0400

    dm cache: fix writethrough mode quiescing in cache_map

    Commit 2ee57d58735 ("dm cache: add passthrough mode") inadvertently
    removed the deferred set reference that was taken in cache_map()'s
    writethrough mode support.  Restore taking this reference.

    This issue was found with code inspection.

    Signed-off-by: Mike Snitzer <snitzer@redhat.com>
    Acked-by: Joe Thornber <ejt@redhat.com>
    Cc: stable@vger.kernel.org # 3.13+

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/dm-cache-target.c b/drivers/md/dm-cache-target.c
index cae7b57..eb0e001 100644
--- a/drivers/md/dm-cache-target.c
+++ b/drivers/md/dm-cache-target.c
@@ -2487,6 +2487,7 @@ static int cache_map(struct dm_target *ti, struct bio *bio)
 
   } else {
    inc_hit_counter(cache, bio);
+   pb->all_io_entry = dm_deferred_entry_inc(cache->all_io_ds);
 
    if (bio_data_dir(bio) == WRITE && writethrough_mode(&cache->features) &&
        !is_dirty(cache, lookup_result.cblock))
-- 
1.7.1